src/gamma.jl

```diff
@@ -590,24 +590,46 @@ lgamma_r
     lfactorial(x)
 
 Compute the logarithmic factorial of a nonnegative integer `x`.
-Equivalent to [`lgamma`](@ref) of `x + 1`, but `lgamma` extends this function
+Equivalent to [`loggamma`](@ref) of `x + 1`, but `loggamma` extends this function
 to non-integer `x`.
 """
-lfactorial(x::Integer) = x < 0 ? throw(DomainError(x, "`x` must be non-negative.")) : lgamma(x + oneunit(x))
+lfactorial(x::Integer) = x < 0 ? throw(DomainError(x, "`x` must be non-negative.")) : loggamma(x + oneunit(x))
 Base.@deprecate lfact lfactorial
 
 """
-    lgamma(x)
+    logabsgamma(x)
 
-Compute the logarithm of the absolute value of [`gamma`](@ref) for
-[`Real`](@ref) `x`, while for [`Complex`](@ref) `x` compute the
-principal branch cut of the logarithm of `gamma(x)` (defined for negative `real(x)`
-by analytic continuation from positive `real(x)`).
+Compute the logarithm of absolute value of [`gamma`](@ref) for
+[`Real`](@ref) `x`.
 """
-function lgamma end
+function logabsgamma end
+logabsgamma(x::Real) = lgamma_r(x)
+logabsgamma(x::Number) = loggamma(x)
 
+"""
+    loggamma(x)
+
+Computes the logarithm of [`gamma`](@ref) for given `x`
+"""
+function loggamma end
+
+
+function loggamma(x::T) where {T<:Union{Float64,Float32}}
+    y, s = lgamma_r(x)
+    println(s)
+    s < 0.0 && throw(DomainError(x, "`gamma(x)` must be non-negative"))
+    return y
+end
+
+function loggamma(x::Real)
+    y, s = lgamma_r(x)
+    s < 0.0 && throw(DomainError(x, "`gamma(x)` must be non-negative"))
+    return y
+end
+loggamma(x::Number) = loggamma(float(x))
+Base.@deprecate lgamma loggamma
```

Comment:
```suggestion
Base.@deprecate lgamma(x::Real) logabsgamma(x)
Base.@deprecate lgamma(x::Number) loggamma(x)
```
and move to deprecated.jl